core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/HBaseCredentialsManager.scala

```diff
@@ -149,8 +151,24 @@ final class HBaseCredentialsManager private() extends Logging {
     }
   }
 
-  private def getNewToken(conf: Configuration): TokenInfo = {
-    val token = TokenUtil.obtainToken(conf)
+  private def getNewToken(conf: Configuration, currentToken: Token[_ <: TokenIdentifier]): TokenInfo = {
+    val token = {
+      if (null != currentToken) {
```

Comment:
TODO: Need to add comments indicating why we need this and how this works.
It is relying on a bit of obscure details - the issue and the fix.